### DISCUSSION TITLE
layout

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1349] [General]
I'm trying to implement some layouts, and there are some tricky places in NiceGui,Here is an example

```py
with ui.row().classes('flex-row'):
    with ui.element('div').classes('flex flex-row container p-2 bg-slate-100 gap-4'):
        ui.button('button1').classes('basis-1/3')
        ui.button('button2').classes('basis-1/2')
    with ui.element('div').classes('flex flex-row container p-2 bg-blue-100 gap-4'):
        ui.button('button1').classes('basis-1/3')
        ui.button('button2').classes('basis-1/2')
```   
the two div don't go in a row .
![微信截图_20230806233215](https://github.com/zauberzeug/nicegui/assets/25096045/98732a22-0f21-427c-8342-611e0c12dbc7)

The user have to test a lot of TailWindCss for every elements to achieve the desire effect .

I thought such layout in the picture may be very easy to achive and dont take much time
![aaa](https://github.com/zauberzeug/nicegui/assets/25096045/b848fd41-158c-4bc2-ae74-a219096907c4)
,but It might takes some time to achieve the desired effect.

If we can provide some default parameter which  could adapt to the common use cases ,and also keep the ablity to adjust the effect with tailwindcss,but only adjust it when necessary
May be It's my problem,I don't have much experience in using CSS,HTML,JS...

**Comment by v479038280** [comment]
> I think this library is a good starting point for people learning web development who already have knowledge of python as it gives complete customisability at the expense of learning how to use CSS etc. I was just like you a couple weeks ago and a bit clueless and while I understand why you'd want less of a learning curve its worth it to learn things so you can transfer skills and as well it leads to quicker development in the future, instead of clueless box making because a library lets you. I made what you asked for in less that 2 minutes and most of it was testing sizes. I think it would be good if you looked into the docs a bit more and learn a bit of CSS.

I agree with you.
It seems the ui.row() didn't work in the example above

**Reply by v479038280** [reply]
Mine,it shoud comes with the no-warp as @falkoschindler said.

**Comment by falkoschindler** [comment]
@v479038280 Regarding your specific example: You might be looking for `with ui.row().classes('no-wrap')` to prevent the row from wrapping.

Apart from that I agree with @Smug246: Since NiceGUI is based on web technologies like HTML, CSS and JavaScript - and provides the flexibility to _use_ them to change its default behavior -, it is very worthwhile to dive into them. You can start with a CSS property here and a Tailwind class there and dig deeper as you gain more and more experience.

**Reply by v479038280** [reply]
Thank you, you are right
